src/dfx/src/commands/start.rs

```diff
@@ -264,6 +265,10 @@ pub fn exec(
     }
     local_server_descriptor.describe(env.get_logger());
 
+    // Get the original background flag set by the user from the command arguments.
+    // Get it from the environment variable as the `--background` flag will be ignored by the send_background() method.
+    let original_background = std::env::var("original_background").is_ok();
```

Comment:
We can define this in the clap definition, but still use an environment variable:

```
#[arg(env = "DFX_RUNNING_IN_BACKGROUND", hide = true)]
bool running_in_background;
```

This way we're declaring it with the rest of the CLI inputs, but it still won't show up in help or affect the shell completion scripts.

Reply:
Fixed at https://github.com/dfinity/sdk/pull/3946/commits/060b25638ae42ecbf2c5854962084a6c5851781e